src/RectangleA.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Maman 12 - Class RectangleA.
+ * The RectangleA class represents a 2D axis aligned integral rectangles.
+ *
+ * @author Noa Amir
+ * @version 2022a
+ */
+
+public class RectangleA {
+    // 1. instance variables
+    private int _width;
+    private int _height;
+    private Point _pointSW;
+
+    // 2. constructors
+
+    /**
+     * First constructor for objects of class RectangleA Constructs a new rectangle with the specified width, height, and it's south-west corner is (0,0)
+     *
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleA(int w, int h) {
+        _pointSW = new Point(0, 0);
+
+        if (w < 1)
+            _width = 1; // TODO: Should I do final DEFAULT_WIDTH = 1
+        else
+            _width = w;
+
+        if (h < 1)
+            _height = 1; // TODO: Should I do final DEFAULT_HEIGHT = 1
+        else
+            _height = h;
+    }
+
+    /**
+     * Second constructor for objects of class RectangleA Construct a new rectangle with the specified width, height and south-west vertex
+     *
+     * @param p - south-western vertex
+     * @param w - The rectangle width
+     * @param h - The rectangle height
+     */
+    public RectangleA(Point p, int w, int h) {
+        _pointSW = new Point(p);
+
+        if (w < 1)
+            _width = 1;
+        else
+            _width = w;
+
+        if (h < 1)
+            _height = 1;
+        else
+            _height = h;
+    }
+
+    /**
+     * Third constructor for objects of class RectangleA Construct a new rectangle with the specified south-west vertex and north-east vertex
+     *
+     * @param sw - south-western vertex
+     * @param ne - north-eastern vertex
+     */
+    public RectangleA(Point sw, Point ne) {
+        _width = ne.getX() - sw.getX();
+        _height = ne.getY() - sw.getY();
+        _pointSW = new Point(sw);
+    }
+
+    /**
+     * Copy constructor for objects of class RectangleA Constructs a rectangle using another rectangle
+     *
+     * @param r - The rectangle from which to construct the new object
+     */
+    public RectangleA(RectangleA r) {
+        _width = r._width;
+        _height = r._height;
+        _pointSW = new Point(r._pointSW);
+    }
+
+    // 3. getters and setters
+
+    /**
+     * Returns the width of the rectangle
+     *
+     * @return the width of the rectangle
+     */
+    public int getWidth() {
+        return _width;
+    }
+
+    /**
+     * Returns the height of the rectangle
+     *
+     * @return the height of the rectangle
+     */
+    public int getHeight() {
+        return _height;
+    }
+
+    /**
+     * Returns the South-West point of the rectangle
+     *
+     * @return a copy of the s-w point of the rectangle
+     */
+    public Point getPointSW() {
+        return new Point(_pointSW);
```

Comment:
?

Reply:
To prevent aliasing.
If I'll do return _pointSW, I'll return the address in heap of the private variable of the class.
Than, it will get to the main and will be possible to edit \ accessible to all.
Break encapsulation!